src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -45,4 +45,10 @@
      * @throws IOException
      */
     Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException;
+
+    /**
+     * Validate the query.
+     * @throws ElasticsearchException if the query is not valid, e.g. fields that are needed to create the lucene query are missing.
+     */
+    void validate() throws ElasticsearchException;
```

Comment:
So you suggest I change signature to `ActionRequestValidationException validate()` and don't throw exceptions, but instead add meaningful error messages to that exception, right? Will change that then, can't use Preconditions then because they throw RuntimeExceptions, but thats not a big problem.